.github/workflows/ci.yml

```diff
@@ -47,19 +47,50 @@ jobs:
         run: |
           go test ./... -short
   integration-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        git-version:
+          - 2.20.0 # oldest supported version
+          - 2.22.5
+          - 2.23.0 # first version with switch and restore
+          - 2.30.8
+          - 2.39.2
     runs-on: ubuntu-latest
-    name: "Integration Tests"
+    name: "Integration Tests - git ${{matrix.git-version}}"
     env:
       GOFLAGS: -mod=vendor
     steps:
+      - name: Restore Git cache
+        id: cache-git-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ~/git-${{matrix.git-version}}
+          key: ${{runner.os}}-git-${{matrix.git-version}}
+      - name: Install Git dependencies
+        run: sudo apt-get install --no-install-recommends -y build-essential ca-certificates curl gettext libcurl4-gnutls-dev libexpat1-dev libssl-dev libz-dev openssl
+      - name: Build Git ${{matrix.git-version}}
+        if: steps.cache-git-restore.outputs.cache-hit != 'true'
+        run: >
+          curl -sL "https://mirrors.edge.kernel.org/pub/software/scm/git/git-${{matrix.git-version}}.tar.xz" -o - | tar xJ -C "$HOME"
+          && cd "$HOME/git-${{matrix.git-version}}"
+          && ./configure
+          && make -j8
+      - name: Install Git ${{matrix.git-version}}
+        run: sudo make -C "$HOME/git-${{matrix.git-version}}" -j8 install
```

Comment:
should we also restrict this step to only run if there was a cache miss?

Reply:
Since `/usr/local/` is not cached, `make install` must be run with or without cache.

This step takes very little time.
<img width="1225" alt="スクリーンショット 2023-03-22 22 50 33" src="https://user-images.githubusercontent.com/10097437/226925395-bcfa77a1-dec2-4ff9-8b1d-b05fc20e7674.png">